### DISCUSSION TITLE
분산 데이터 처리에서 sort의 결과는 write후 read할 때는 어떻게 되나요?

### DISCUSSION THREAD

**Discussion by qlqhqo2341** [NMP-Study/spark2023#27] [Q&A] [answered]
- 데이터가 파티션으로 나누어 저장되는데 orderBy, sort를 하게 된다면 파티셔닝에 추가적인 효과를 가져오는 걸까?

**Comment by KimYunsang-v** [comment, marked as answer]
### 확인 결과

- 파티션이 여러 개 일 때 orderBy or sort 후 저장한다면 파티션 사이의 순서는 보장되지 않는다.
- 하나의 파티션 내에서는 순서가 보장됨.

**Comment by jeongwon-p** [comment]
다들 그래서 정렬이 보장되야하면 단일 파티션으로 하는거 같군요.
스파크 버전 버그라는 말도 있는거 같아요!
https://stackoverflow.com/questions/71154277/different-sort-results-after-coalesce1-vs-repartition1

**Comment by qlqhqo2341** [comment]
나름 탐구해봤는데 정렬이 꼭 필요하다면 단일 파티션이 답인 것 같아요...ㅋㅋ
제플린의 `jhyee/discussions_27` 에 테스트 코드도 남겨뒀습니다!
